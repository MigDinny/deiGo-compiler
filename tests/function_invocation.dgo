package main;

var a, b, x, y, z int;

func main() {
    if (x && y || z){
        for {
            /*         infinite loop       */
            trap = infinite(a || b, x, y, z)
            fmt.Println(trap)
        }
        fmt.Println("Depois do for")
    }
    fmt.Println("Depois do if")
} // No new Line at the End