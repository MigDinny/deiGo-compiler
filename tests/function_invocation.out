Program
..VarDecl
....Int
....Id(a)
..VarDecl
....Int
....Id(b)
..VarDecl
....Int
....Id(x)
..VarDecl
....Int
....Id(y)
..VarDecl
....Int
....Id(z)
..FuncDecl
....FuncHeader
......Id(main)
......FuncParams
....FuncBody
......If
........Or
..........And
............Id(x)
............Id(y)
..........Id(z)
........Block
..........For
............Block
..............Assign
................Id(trap)
................Call
..................Id(infinite)
..................Or
....................Id(a)
....................Id(b)
..................Id(x)
..................Id(y)
..................Id(z)
..............Print
................Id(trap)
..........Print
............StrLit("Depois do for")
........Block
......Print
........StrLit("Depois do if")